src/games/strategy/engine/framework/mapDownload/DownloadRunnable.java

```diff
@@ -52,26 +59,45 @@ public void setError(final String error) {
 
   @Override
   public void run() {
+    if (urlString.startsWith("http")) {
+      downloadFile();
+    } else {
+      readLocalFile();
+    }
+  }
+
+  private void downloadFile() {
     URL url;
+
     try {
-      // System.out.println(System.getProperty("http.proxyHost"));
-      // System.out.println(System.getProperty("http.proxyPort"));
       url = new URL(urlString.trim());
     } catch (final MalformedURLException e1) {
       error = "invalid url";
       return;
     }
+    try {
```

Comment:
Surprisingly as well none of these resources need a "close()" after they are created, hence why there are no java7 autoclose blocks here.